dasharo-compatibility/usb-type-c-semiauto.robot

```diff
@@ -0,0 +1,605 @@
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    UTC011.005 not supported
+    Skip If    not ${PLATFORM_SLEEP_TYPE_SELECTABLE}    UTC011.005 not supported
+    Pause Execution In Console    Please make sure the docking station is disconnected and press ENTER
+    Set Platform Sleep Type    S0ix
+    Docking Station Detection After Suspend Then Hotplug (Ubuntu 22.04)    WL-UMD05 Pro Rev.C1    S0ix
+
+UTC025.006 Docking station detection after suspend then hotplug (Ubuntu 22.04) (S3) (WL-UMD05 Pro Rev.C1)
+    [Documentation]    Check whether the DUT properly detects the docking station
+    ...    after reboot.
+    Skip If    not ${DOCKING_STATION_DETECT_SUPPORT}    UTC011.006 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    UTC011.006 not supported
+    Skip If    not ${PLATFORM_SLEEP_TYPE_SELECTABLE}    UTC011.006 not supported
+    Pause Execution In Console    Please make sure the docking station is disconnected and press ENTER
+    Set Platform Sleep Type    S3
+    Docking Station Detection After Suspend Then Hotplug (Ubuntu 22.04)    WL-UMD05 Pro Rev.C1    S3
+
+UTC022.003 Docking station detection after coldboot then hotplug (Ubuntu 22.04) (WL-UG69PD2 Rev.A1)
+    [Documentation]    Check whether he DUT properly detects the docking station
+    ...    after coldboot.
+    Skip If    not ${DOCKING_STATION_DETECT_SUPPORT}    UTC008.003 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    UTC008.003 not supported
+    Docking Station Detection After Coldboot Then Hotplug (Ubuntu 22.04)    WL-UG69PD2 Rev.A1
+
+UTC023.003 Docking station detection after warmboot then hotplug (Ubuntu 22.04) (WL-UG69PD2 Rev.A1)
+    [Documentation]    Check whether he DUT properly detects the docking station
+    ...    after warmboot.
+    Skip If    not ${DOCKING_STATION_DETECT_SUPPORT}    UTC009.003 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    UTC009.003 not supported
+    Docking Station Detection After Warmboot Then Hotplug (Ubuntu 22.04)    WL-UG69PD2 Rev.A1
+
+UTC024.003 Docking station detection after reboot then hotplug (Ubuntu 22.04) (WL-UG69PD2 Rev.A1)
+    [Documentation]    Check whether the DUT properly detects the docking station
+    ...    after reboot.
+    Skip If    not ${DOCKING_STATION_DETECT_SUPPORT}    UTC010.003 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    UTC010.003 not supported
+    Docking Station Detection After Reboot Then Hotplug (Ubuntu 22.04)    WL-UG69PD2 Rev.A1
+
+UTC025.007 Docking station detection after suspend then hotplug (Ubuntu 22.04) (WL-UG69PD2 Rev.A1)
+    [Documentation]    Check whether the DUT properly detects the docking station
+    ...    after reboot.
+    Skip If    not ${DOCKING_STATION_DETECT_SUPPORT}    UTC011.007 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    UTC011.007 not supported
+    Skip If    ${PLATFORM_SLEEP_TYPE_SELECTABLE}    UTC011.007 not supported
```

Comment:
```suggestion
    Skip If    not ${PLATFORM_SLEEP_TYPE_SELECTABLE}    UTC011.007 not supported
```

Is this how it should be? I am not sure why the condition is even here in the first place. Does this test need to run only on platforms which do not have the selectable sleep type?

Reply:
No, of course you're right. I missed this

Reply:
Okay, I re-read these and I think there are generic test case for `Suspend` when sleep type is not selectable, and separate `S0ix` and `S3` cases for when sleep type IS selectable. Probably not ideal, ideally we'd specify in board config which sleep types are available and choose the right test cases automatically?

Reply:
@miczyg1 As we discussed with @macpijan on Matrix, I guess we may keep this as is for now and consider refactoring the sleep mode tests in the future, after the release.

Reply:
Sure @mkopec just please create tracking issue for that.

Reply:
@macpijan added an issue: https://github.com/Dasharo/open-source-firmware-validation/issues/163

Reply:
So is my code suggestion correct? Not sure what should happen here/what was the intention of these skip conditions

Reply:
@miczyg1 No. We skip the generic test case (like the one you commented on) on laptops without sleep type switch. We skip the S3 and S0ix test cases (indicated in case name) on platforms without the sleep type switch.